src/main/cpp/utils/logging/signals/DragonDataLogger.cpp

```diff
@@ -33,219 +34,38 @@ void DragonDataLogger::LogBoolData(uint64_t timestamp, DragonDataLogger::BoolSig
     auto dataMgr = DragonDataLoggerMgr::GetInstance();
     if (dataMgr != nullptr)
     {
-        switch (signalID)
+        std::string nameprefix = std::string(magic_enum::enum_name(signalID));
```

Comment:
Not really a fan of doing string manipulations every time we log.  Why not create a map so, you only have to do it once?